### PR TITLE
generate cocos-version.h

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/cocos-engine/issues/12851

### Changelog

* generate cocos-version.h in CI

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/jswrapper/PrivateObject.h [hunk 1]
```diff
@@ -131,33 +131,24 @@ class CCIntrusivePtrPrivateObject final : public TypedPrivateObject<T> {
...
 };
 
-template <typename T>
-inline typename std::enable_if_t<std::is_destructible<T>::value, void> cctryDelete(T *t) {
-    delete t;
-}
-template <typename T>
-inline typename std::enable_if_t<!std::is_destructible<T>::value, void> cctryDelete(T *t) {
-}
-
```
Comment (minggo) [inline native/cocos/bindings/jswrapper/PrivateObject.h]: Are these changed related with this PR?

Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/PrivateObject.h]: These are small changes and descripted in changelog

Comment (minggo) [reply native/cocos/bindings/jswrapper/PrivateObject.h]: Please do one thing in a PR.

Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/PrivateObject.h]: fixed

**Review by minggo — APPROVED** [review]
